### PR TITLE
Initial commit of expanded System.Runtime

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**

### CODE REVIEW

**Comment by KrzysztofCwalina** [comment]
@weshaggard, should I merge it?

File: src/System.Runtime/ref/System.Runtime.csproj [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <AssemblyVersion>4.1.0.0</AssemblyVersion>
+    <AssemblyVersion>4.2.0.0</AssemblyVersion>
     <OutputType>Library</OutputType>
     <IsCoreAssembly>true</IsCoreAssembly>
     <PackageTargetFramework>netstandard1.5</PackageTargetFramework>
```
Comment (weshaggard) [inline src/System.Runtime/ref/System.Runtime.csproj]: We should bump this to NS1.6

**Comment by weshaggard** [comment]
>   public sealed partial class SafeFileHandle : Microsoft.Win32.SafeHandles.SafeHandleZeroOrMinusOneIsInvalid {

Given you are adding types to this contract that exist in other contracts no one will be able to use this new contract until those other contracts are actually updated as well. 

---
Refers to: src/System.Runtime/ref/System.Runtime.cs:11 in 355c1a4. [](commit_id = 355c1a486c44e389ffa7141a1c83bc385c9bce57, deletion_comment = False)

**Comment by weshaggard** [comment]
>     [System.Runtime.ConstrainedExecution.ReliabilityContractAttribute((System.Runtime.ConstrainedExecution.Consistency)(3), (System.Runtime.ConstrainedExecution.Cer)(1))]

Can we please remove the Reliability attributes from the reference assembly?

---
Refers to: src/System.Runtime/ref/System.Runtime.cs:19 in 355c1a4. [](commit_id = 355c1a486c44e389ffa7141a1c83bc385c9bce57, deletion_comment = False)

**Comment by weshaggard** [comment]
>   [System.Runtime.CompilerServices.TypeForwardedFromAttribute("System.Core, Version=2.0.5.0, Culture=Neutral, PublicKeyToken=7cec85d7bea7798e")]

Remove TypeForwardedFromAttributes from ref.

---
Refers to: src/System.Runtime/ref/System.Runtime.cs:45 in 355c1a4. [](commit_id = 355c1a486c44e389ffa7141a1c83bc385c9bce57, deletion_comment = False)

**Comment by weshaggard** [comment]
>   [System.Diagnostics.DebuggerDisplayAttribute("Count = {InnerExceptionCount}")]

Looks like we have a number of attributes that we should probably remove from the ref still.

---
Refers to: src/System.Runtime/ref/System.Runtime.cs:95 in 355c1a4. [](commit_id = 355c1a486c44e389ffa7141a1c83bc385c9bce57, deletion_comment = False)

**Comment by weshaggard** [comment]
>     [System.Runtime.TargetedPatchingOptOutAttribute("Performance critical to inline across NGen image boundaries")]

Remove TargetedPatching attributes. 

---
Refers to: src/System.Runtime/ref/System.Runtime.cs:1232 in 355c1a4. [](commit_id = 355c1a486c44e389ffa7141a1c83bc385c9bce57, deletion_comment = False)

**Comment by weshaggard** [comment]
>     public const double PositiveInfinity = 1.0 / 0.0;

Most of these constants are already in the manual file and I'm surprised this actually compiled with them in both places.  Just so you know why these are defined in the manual file it is because the values generated by the tools were slightly different from the real constant values so we needed to specify them manually exactly like they were in the source for the implementation.

---
Refers to: src/System.Runtime/ref/System.Runtime.cs:1633 in 355c1a4. [](commit_id = 355c1a486c44e389ffa7141a1c83bc385c9bce57, deletion_comment = False)

**Comment by weshaggard** [comment]
>   public partial class Random {

Types like this were in System.Runtime.Extensions and as best I can tell they aren't actually needed in the public API surface of System.Runtime. Do you intend to move them all into System.Runtime still or are you going to keep them in System.Runtime.Extensions or other existing contracts?

---
Refers to: src/System.Runtime/ref/System.Runtime.cs:2672 in 355c1a4. [](commit_id = 355c1a486c44e389ffa7141a1c83bc385c9bce57, deletion_comment = False)

**Comment by weshaggard** [comment]
>   public sealed partial class DBNull : System.IConvertible, System.Runtime.Serialization.ISerializable {

FYI: This is in a higher level contract System.Data.Common.

---
Refers to: src/System.Runtime/ref/System.Runtime.cs:1385 in 355c1a4. [](commit_id = 355c1a486c44e389ffa7141a1c83bc385c9bce57, deletion_comment = False)

**Comment by weshaggard** [comment]
>   public partial class Dictionary<TKey, TValue> : System.Collections.Generic.ICollection<System.Collections.Generic.KeyValuePair<TKey, TValue>>, System.Collections.Generic.IDictionary<TKey, TValue>, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<TKey, TValue>>, System.Collections.Generic.IReadOnlyCollection<System.Collections.Generic.KeyValuePair<TKey, TValue>>, System.Collections.Generic.IReadOnlyDictionary<TKey, TValue>, System.Collections.ICollection, System.Collections.IDictionary, System.Collections.IEnumerable, System.Runtime.Serialization.IDeserializationCallback, System.Runtime.Serialization.ISerializable {

All of System.Collections is also going into System.Runtime?

---
Refers to: src/System.Runtime/ref/System.Runtime.cs:4174 in 355c1a4. [](commit_id = 355c1a486c44e389ffa7141a1c83bc385c9bce57, deletion_comment = False)

**Comment by weshaggard** [comment]
>   public partial class FileStream : System.IO.Stream {

Didn't you decide to keep FileStream higher level?

---
Refers to: src/System.Runtime/ref/System.Runtime.cs:5176 in 355c1a4. [](commit_id = 355c1a486c44e389ffa7141a1c83bc385c9bce57, deletion_comment = False)

**Comment by weshaggard** [comment]
>   public sealed partial class SerializationInfo {

You decided to put all the serialization stuff in System.Runtime?

---
Refers to: src/System.Runtime/ref/System.Runtime.cs:7175 in 355c1a4. [](commit_id = 355c1a486c44e389ffa7141a1c83bc385c9bce57, deletion_comment = False)

**Comment by weshaggard** [comment]
>   public sealed partial class SecureString : System.IDisposable {

Why is SecureString in System.Runtime?

---
Refers to: src/System.Runtime/ref/System.Runtime.cs:7281 in 355c1a4. [](commit_id = 355c1a486c44e389ffa7141a1c83bc385c9bce57, deletion_comment = False)

**Comment by weshaggard** [comment]
@KrzysztofCwalina I would like to review this a little further at one of our design reviews to understand better some of your thoughts on what you are adding here.

**Comment by KrzysztofCwalina** [comment]
I removed the attributes (in fxdac), DbNull, SecureString.

Yes, the existing contracts will need to be changed (to type forward), but I was planing to do it contract by contract (at least initially).

I removed the constants from double and single. I am not sure why it compiled yesterday in my office. It did not compile today at home.

I would like to talk about FileStream and serialization separately. There is no way to refactor them without significant API removals. 

And yes, let's meet and review the rest of the list.

**Comment by weshaggard** [comment]
@KrzysztofCwalina we now have CI for this branch so please fix up the build and then go ahead and merge and we can review and make further tweaks later.

**Comment by KrzysztofCwalina** [comment]
I will close the PR. It's not even close to compiling. I am not sure how we (Wes and I) compiled it in my office the other day. Also, as Wes said, we should review the APIs.

**Pull request closed without merging**